core-bundle/contao/drivers/DC_Table.php

```diff
@@ -3205,109 +3205,105 @@ protected function save($varValue)
 
 	protected function submit()
 	{
-		if (empty($this->arrSubmit) || Input::post('FORM_SUBMIT') != $this->strTable)
-		{
-			return;
-		}
-
-		if ($this->noReload)
+		if (Input::post('FORM_SUBMIT') != $this->strTable)
 		{
-			// Data should not be submitted due to validation errors
-			$this->arrSubmit = array();
-
 			return;
 		}
 
 		$arrValues = $this->arrSubmit;
 		$this->arrSubmit = array();
 
-		// Call onbeforesubmit_callback
-		if (\is_array($GLOBALS['TL_DCA'][$this->strTable]['config']['onbeforesubmit_callback'] ?? null))
+		if (!$this->noReload && !empty($arrValues))
 		{
-			foreach ($GLOBALS['TL_DCA'][$this->strTable]['config']['onbeforesubmit_callback'] as $callback)
+			$arrValues['tstamp'] = time();
+
+			if ($GLOBALS['TL_DCA'][$this->strTable]['config']['dynamicPtable'] ?? null)
 			{
-				try
+				$arrValues['ptable'] = $this->ptable;
+			}
+
+			// Trigger the onbeforesubmit_callback
+			if (\is_array($GLOBALS['TL_DCA'][$this->strTable]['config']['onbeforesubmit_callback'] ?? null))
+			{
+				foreach ($GLOBALS['TL_DCA'][$this->strTable]['config']['onbeforesubmit_callback'] as $callback)
 				{
-					if (\is_array($callback))
+					try
 					{
-						$this->import($callback[0]);
-						$arrValues = $this->{$callback[0]}->{$callback[1]}($arrValues, $this);
+						if (\is_array($callback))
+						{
+							$this->import($callback[0]);
+							$arrValues = $this->{$callback[0]}->{$callback[1]}($arrValues, $this);
+						}
+						elseif (\is_callable($callback))
+						{
+							$arrValues = $callback($arrValues, $this);
+						}
 					}
-					elseif (\is_callable($callback))
+					catch (\Exception $e)
 					{
-						$arrValues = $callback($arrValues, $this);
+						$this->noReload = true;
+						Message::addError($e->getMessage());
+
+						break;
 					}
 
 					if (!\is_array($arrValues))
```

Comment:
Shouldn‘t this `if` condition be outside the `foreach` loop?

Reply:
As `$arrValues` is used as an argument, in case of an error, the next `onbeforesubmit` callback would then potentially fail. Then you might not see the exception.